homeassistant/components/sensor/template.py

```diff
@@ -5,46 +5,43 @@
 https://home-assistant.io/components/sensor.template/
 """
 import logging
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
 
-from homeassistant.components.sensor import ENTITY_ID_FORMAT
+from homeassistant.components.sensor import ENTITY_ID_FORMAT, PLATFORM_SCHEMA
 from homeassistant.const import (
     ATTR_FRIENDLY_NAME, ATTR_UNIT_OF_MEASUREMENT, CONF_VALUE_TEMPLATE,
     ATTR_ENTITY_ID, MATCH_ALL)
 from homeassistant.exceptions import TemplateError
 from homeassistant.helpers.entity import Entity, generate_entity_id
 from homeassistant.helpers import template
 from homeassistant.helpers.event import track_state_change
-from homeassistant.util import slugify
 
 _LOGGER = logging.getLogger(__name__)
 CONF_SENSORS = 'sensors'
 
+SENSOR_SCHEMA = vol.Schema({
+    cv.slug: vol.Schema({
+        vol.Required(CONF_VALUE_TEMPLATE): cv.template,
+        vol.Optional(ATTR_FRIENDLY_NAME): cv.string,
+        vol.Optional(ATTR_UNIT_OF_MEASUREMENT): vol.Coerce(str),
+        vol.Optional(ATTR_ENTITY_ID): cv.entity_ids})
+    })
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SENSORS): vol.All(dict, SENSOR_SCHEMA),
```

Comment:
You don't have to validate that it is a `dict` as the `SENSOR_SCHEMA` already does this. However, you do want to verify that each key is a slug while the value match SENSOR SCHEMA:

```python
    vol.Required(CONF_SENSORS): {cv.slug: SENSOR_SCHEMA},
```